api/app/src/main/kotlin/packit/service/BasePacketService.kt

```diff
@@ -74,4 +67,33 @@ class BasePacketService(
     {
         return this.joinToString("") { "%02x".format(it) }
     }
+
+    override fun getMetadataBy(id: String): PacketMetadata
+    {
+        return outpackServerClient.getMetadataById(id)
+            ?: throw PackitException("doesNotExist", HttpStatus.NOT_FOUND)
+    }
+
+    override fun getFileByHash(hash: String, inline: Boolean, filename: String): Pair<InputStreamResource, HttpHeaders>
+    {
+        val response = outpackServerClient.getFileByHash(hash)
+
+        if (response?.first == null || response.first.isEmpty())
+        {
+            throw PackitException("doesNotExist", HttpStatus.NOT_FOUND)
+        }
+
+        val inputStream = response.first.toString().byteInputStream()
+
+        val inputStreamResource = InputStreamResource(inputStream)
```

Comment:
This conversion is causing the problem with rendering. Just change the return type of this method to `ByteArrayResource` and then here do:

```suggestion
        val byteArrayResource = ByteArrayResource(response.first)
```